client/connectivity/connectivity.go

```diff
@@ -230,6 +230,13 @@ func (c *circle) newPeer(ctx context.Context, p *pb.Peer) *Peer {
 		grpc.WithContextDialer(c.dialPeer),
 		grpc.WithTransportCredentials(credentials.NewTLS(c.keyPair.TLSConfigForServerClient(p.GetName()))),
 		keepaliveOption,
+		// gRPC should keep a channel open to all endpoints, since UDP hole-punching
+		// between two clients may only work if they are trying to connect to each
+		// other via UDP. Hence, even if one client can connect to the other via TCP,
+		// and this method of communication is preferred, the client should still
+		// also connect using UDP; if not, the other client may not be able to
+		// connect to the first client at all.
+		grpc.WithBalancerName(BalancerName),
```

Comment:
WithBalancerName is deprecated, we should switch to ServiceConfig at some point.

Reply:
I wanted to, but could not find any documentation on how to use a non-standard Balancer with a ServiceConfig... I could find how to pass a balancer policy string but that's *also* deprecated, and when giving a balancer config I couldn't figure out how to pass the balancer name as a string...

Reply:
My hope was that when they actually do remove `WithBalancerName`, at least they'd document the alternative properly, so we could switch then.

Reply:
It took me a while to figure this out (looks like it's involved to return a service config from a manual resolver like we use, I filed https://github.com/grpc/grpc-go/issues/4618 for that).

But simply this should work:
  grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [ { "rufs-peer": {} } ]}`),

(based on the second example in https://github.com/grpc/grpc/blob/master/doc/service_config.md#example and the last paragraph of https://github.com/grpc/grpc/blob/master/doc/service_config.md#format)

Reply:
That works indeed! I missed, from the proto files I inspected, how the JSON key should be the name of the balancer. Thanks, PR is updated.